multiapps-controller-core/src/main/java/org/cloudfoundry/multiapps/controller/core/validators/parameters/RoutesValidator.java

```diff
@@ -71,22 +74,42 @@ public Object attemptToCorrect(Object routes, final Map<String, Object> context)
         }
 
         return routesList.stream()
+                         .map(originalElement -> getUpdatedAllRouteElements(originalElement, context))
                          .map(this::attemptToCorrectParameterMap)
                          .collect(Collectors.toList());
     }
 
-    private Map<String, Object> attemptToCorrectParameterMap(Map<String, Object> originalElem) {
+    private Map<String, Object> getUpdatedAllRouteElements(Map<String, Object> routeElement, Map<String, Object> context) {
+        Map<String, Object> updatedAllRouteElements = new HashMap<>(routeElement);
+        Set<String> relatedParameters = getRelatedParameterNames();
+        Map<String, Object> globalParametersValues = relatedParameters.stream()
+                                                                      .filter(relatedParameter -> isGlobalParameterValueOverridden(relatedParameter,
+                                                                                                                                   routeElement,
+                                                                                                                                   context))
+                                                                      .collect(Collectors.toMap(Function.identity(), context::get));
+        updatedAllRouteElements.putAll(globalParametersValues);
+        return updatedAllRouteElements;
+    }
+
+    private boolean isGlobalParameterValueOverridden(String relatedParameter, Map<String, Object> routeElement,
```

Comment:
doesn't this check the opposite of what the method name implies? my mistake - you can negate the check or name it isGlobalParameterValueInAction or isGlobalParameterValueNotOverridden :D